src/modules/booking/components/booking-time-select-navigation/booking-time-select-navigation.tsx

```diff
@@ -0,0 +1,54 @@
+import { useMemo, type FC } from 'react';
+import { useSwiper } from 'swiper/react';
+// components
+import { Typography } from '@/modules/core/components/typogrpahy';
+import { Button } from '@/modules/core/components/button';
+// utils
+import { generateDates } from '@/modules/core/utils/date.utils';
+// type
+import type { BookingTimeSelectNavigationProps } from './booking-time-select.interface';
+// style
+import styles from './booking-time-select-navigation.module.scss';
+import { format } from 'date-fns';
+
+const selectDate = (
+  <Typography variant='body1' className={styles.selectDate}>
+    Select a date
+  </Typography>
+);
+
+export const BookingTimeSelectNavigation: FC<
+  BookingTimeSelectNavigationProps
+> = ({ selectedDay }) => {
+  const swiper = useSwiper();
+  const dates = generateDates();
+  const selectedDate = useMemo(() => {
+    if (selectedDay) {
+      return dates.find((item) => {
+        return selectedDay === item;
```

Comment:
I suggest you to remove second return and change if a bit
`if (!selectedDay) { return null;}`
In that case we handle exception first and then work with the data. In my personal opinion int looks better.